astropy/units/quantity.py

```diff
@@ -1138,6 +1138,13 @@ def get_virtual_unit_attribute():
     # DeprecationWarnings on any error, which is distracting, and does not
     # deal well with structured arrays (nor does the ufunc).
     def __eq__(self, other):
+        if (
+            hasattr(other, "unit")
+            and hasattr(other.unit, "physical_type")
+            and (other.unit.physical_type) != self.unit.physical_type
```

Comment:
Is this faster?

```suggestion
            and getattr(other.unit, "physical_type", True) != getattr(self.unit, "physical_type", False)
```

Reply:
It *is* faster. Running my script again:
my patch:
```
'data==data2' took 5.7 s (or 11.4 us/iteration)
'data!=data2' took 5.6 s (or 11.3 us/iteration)
'data==data3' took 5.2 s (or 10.4 us/iteration)
'data!=data3' took 5.4 s (or 10.8 us/iteration)
```
with your suggestion
```
'data==data2' took 4.6 s (or 9.1 us/iteration)
'data!=data2' took 4.5 s (or 9.0 us/iteration)
'data==data3' took 4.7 s (or 9.3 us/iteration)
'data!=data3' took 4.6 s (or 9.3 us/iteration)
```
I'll push this, thank you !